native/crypto/gcm_util.c

```diff
@@ -91,8 +92,17 @@ GCM_JNI_CTX* Create_GCM_JNI_CTX(jbyte* keyBytes, jbyte* ivBytes) {
     return NULL;
   }
 
+  if (ivLength > GCM_IV_LENGTH_IN_BYTES) {
+    ivLength = GCM_IV_LENGTH_IN_BYTES;
+  }
+
   memcpy(ctx->key, keyBytes, GCM_KEY_LENGTH_IN_BYTES);
-  memcpy(ctx->iv, ivBytes, GCM_IV_LENGTH_IN_BYTES);
+  memcpy(ctx->iv, ivBytes, ivLength);
+  if (ivLength < GCM_IV_LENGTH_IN_BYTES) {
+    // pad IV with zeros if provided IV is smaller
+    // this will remove undefined behavior (which worked because the bytes were already the same)
+    memset(ctx->iv+ivLength, 0, GCM_IV_LENGTH_IN_BYTES-ivLength);
```

Comment:
style nit, spaces between the operator 

  ctx->iv + ivLength
  and GCM_IV_LENGTH_IN_BYTES - ivLength